Shrinking of negative Int64 around 0 is making panics in check.go

The issue can be reproduced with following test:

```go
import (
  "fmt"
  "testing"
  
  "github.com/steffnova/go-check"
  "github.com/steffnova/go-check/constraints"
  "github.com/steffnova/go-check/generator"
)

func TestPropertyInt(t *testing.T) {
  predicate := func(x int) error {
    // Predicate fails if x is anything other than 0
    if x != 0 {
      return fmt.Errorf("")
    }
    return nil
  }
  
  check.Check(t, check.Property(
    predicate,
    generator.Int(constraints.Int{Min: -20, Max: 0}),
  ))
}
```

Panic occurs:
```
--- FAIL: TestPropertyInt (0.00s)
panic: interface conversion: interface is nil, not error [recovered]
	panic: interface conversion: interface is nil, not error

goroutine 18 [running]:
testing.tRunner.func1.2(0x1132960, 0xc00011c600)
	/usr/local/go/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000102600)
	/usr/local/go/src/testing/testing.go:1146 +0x4b6
panic(0x1132960, 0xc00011c600)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/steffnova/go-check.Property.func1.1(0x1184e38, 0xc00011c390)
	/test/go-check/property.go:82 +0x73b
github.com/steffnova/go-check.Check(0xc000102600, 0xc00011c360, 0x0, 0x0, 0x0)
	/test/go-check/check.go:37 +0x1e2
github.com/steffnova/go-check/blazing.TestPropertyInt(0xc000102600)
	/go-check/blazing/check_test.go:69 +0xe7
testing.tRunner(0xc000102600, 0x11603c8)
	/usr/local/go/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1238 +0x2b3
FAIL	github.com/steffnova/go-check/blazing	1.597s
FAIL
```

Expected result is for x to be shrunk to -1 as it is the smallest negative number that satisfies the property condition.

--- a/shrinker/int.go
+++ b/shrinker/int.go
@@ -6,38 +6,63 @@ import (
 	"github.com/steffnova/go-check/constraints"
 )
 
-// Int64 is a shrinker for int64. X is the shrinking target and limits are
-// constraints in which x will be shrunk. If x >= 0 it will be shrunk towards
-// limits.Min or 0 whichever is higher. If x < 0 it will be shrunk towards 0
-// or limits.Max, whichever is higher.
+// Int64 is a shrinker for int64. N is the shrinking target and limits are
+// constraints in which n will be shrunk. If n > 0 it will be shrunk towards
+// limits.Min or 0 whichever is higher. If n < 0 it will be shrunk towards 0
+// or limits.Max, whichever is lower. If n == 0 it will be returned with no
+// shrinker as it is converging point in ranges [math.Int64Min, 0] and [0, math.Int64Max]
 func Int64(n int64, limits constraints.Int64) Shrinker {
 	switch {
-	case n >= 0 && limits.Min < 0:
-		limits.Min = 0
-	case n < 0 && limits.Max > 0:
-		limits.Max = 0
+	case n > 0:
+		return int64Positive(n, limits)
+	case n < 0:
+		return int64Negative(n, limits)
+	default:
+		return func(propertyFailed bool) (reflect.Value, Shrinker) {
+			return reflect.ValueOf(int64(0)), nil
+		}
 	}
 
+}
+
+// int64Positive is a shrinker of positive int64 numbers. All numbers
+// are shrunk towards 0 or limits.Min whichever is higher.
+func int64Positive(n int64, limits constraints.Int64) Shrinker {
+	if limits.Min < 0 {
+		limits.Min = 0
+	}
 	return func(propertyFailed bool) (reflect.Value, Shrinker) {
 		switch {
 		case limits.Max == limits.Min:
 			return reflect.ValueOf(n), nil
-		case n >= 0:
-			if propertyFailed {
-				limits.Max = n
-			} else {
-				limits.Min = n + 1
-			}
-			shrinked := limits.Min + (limits.Max-limits.Min)/2
-			return reflect.ValueOf(shrinked), Int64(shrinked, limits)
+		case propertyFailed:
+			limits.Max = n
 		default:
-			if propertyFailed {
-				limits.Min = n
-			} else {
-				limits.Max = n - 1
-			}
-			shrinked := limits.Max - (limits.Max-limits.Min)/2
-			return reflect.ValueOf(shrinked), Int64(shrinked, limits)
+			limits.Min = n + 1
 		}
+
+		shrinked := limits.Max - ((limits.Max-limits.Min)/2 + (limits.Max-limits.Min)%2)
+		return reflect.ValueOf(shrinked), int64Positive(shrinked, limits)
+	}
+}
+
+// int64Negative is a shrinker of negative int64 numbers. All numbers
+// are shrunk towards 0 or limits.Max whichever is lower.
+func int64Negative(n int64, limits constraints.Int64) Shrinker {
+	if limits.Max > 0 {
+		limits.Max = 0
+	}
+	return func(propertyFailed bool) (reflect.Value, Shrinker) {
+		switch {
+		case limits.Max == limits.Min:
+			return reflect.ValueOf(n), nil
+		case propertyFailed:
+			limits.Min = n
+		default:
+			limits.Max = n - 1
+		}
+
+		shrinked := limits.Max - (limits.Max-limits.Min)/2
+		return reflect.ValueOf(shrinked), int64Negative(shrinked, limits)
 	}
 }

